webui/react/src/stores/workspaces.tsx

```diff
@@ -107,3 +108,42 @@ export const useUpdateWorkspace = (): ((
     [updateWorkspaces],
   );
 };
+
+export const useCreateWorkspace = (): ((arg0: V1PostWorkspaceRequest) => Promise<Workspace>) => {
+  const context = useContext(WorkspacesContext);
+  if (context === null) {
+    throw new Error('Attempted to use useCreateWorkspace outside of Workspace Context');
+  }
+  const { updateWorkspaces } = context;
+
+  return useCallback(
+    async (params: V1PostWorkspaceRequest): Promise<Workspace> => {
+      const w = await createWorkspace(params);
+      updateWorkspaces((prev) => Loadable.map(prev, (ws: Workspace[]) => ws.concat([w])));
+      return w;
+    },
+    [updateWorkspaces],
+  );
+};
+
+export const useDeleteWorkspace = (): ((id: number) => Promise<void>) => {
+  const context = useContext(WorkspacesContext);
+  if (context === null) {
+    throw new Error('Attempted to use useDeleteWorkspace outside of Workspace Context');
+  }
+  const { updateWorkspaces } = context;
+
+  return useCallback(
+    async (id: number): Promise<void> => {
+      try {
+        await deleteWorkspace({ id });
```

Comment:
this `await`  has no effect

Reply:
is it needed to wait for possible errors when deleting?

Reply:
Editor suggests it 
`updateWorkspaces()` is a normal func 

<img width="713" alt="image" src="https://user-images.githubusercontent.com/109113616/208203340-36bafaf9-cfbb-4689-8f9b-e695c74c2602.png">

Reply:
oh, that's a different section of this file (inside of `useUpdateWorkspace`), which only updates the local store, so I just pushed a commit to remove async/await there.

Inside of `useDeleteWorkspace` I am going to keep the async/await, because it makes a remote call and if we wanted any return values or additional lines of code, it would need to be async/await.

Reply:
You also don't need a try/catch block since `updateWorkspaces` never throws.

Reply:
✅  changed Promise<void> to void, removed the try-catch,